crates/ruma-events/src/room/message.rs

```diff
@@ -977,7 +977,8 @@ fn get_plain_quote_fallback(original_message: &MessageEvent) -> String {
         MessageType::VerificationRequest(content) => {
             format!("> <{:?}> {}", original_message.sender, content.body)
         }
-    }
+    };
+    start_quoted.replace('\n', "\n> ")
```

Comment:
Why not

```suggestion
    }.replace('\n', "\n> ")
```

+ removing the line above

if `replace` returns a string and doesn't edit in-place, then this'd be better

Reply:
Actually, the extra string allocation can be avoided entirely. Without itertools it's not as easy as `.split('\n').format("\n> ")` on a few `content.body`'s, but I still think it's worth it. @ftilde do you want to create an equivalent helper function and use that here, or would you prefer that work be left to sbd. else as a followup PR?

Reply:
Well, if we really want to avoid allocations it would probably better to
change the helper function entirely to not return part of the reply body
as a string and (for example) construct the whole body at once.

However, I personally think that this optimization is not worth it here,
especially since it will probably hurt readability/simplicity. I highly
doubt that this function is hot in anyone's (most likely client!) code.

Reply:
From a understanding / maintainability perspective I would prefer the newline replacement only happening in the branches that actually can be multiline. And I don't think the `split` + `format` would be harder to read than `replace`. But I'm okay merging this as-is.